### PR TITLE
Replace list of collectors with array in profile collector

### PR DESCRIPTION
**Pull request by javanna**
It simplifies things if we can just use an array to hold the children collectors in InternalProfileCollector, and adjust its constructor to take a varargs argument of collectors

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/profile/query/InternalProfileCollector.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public CollectorResult getCollectorTree() {
     }
 
     private static CollectorResult doGetCollectorTree(InternalProfileCollector collector) {
-        List<CollectorResult> childResults = new ArrayList<>(collector.children.size());
+        List<CollectorResult> childResults = new ArrayList<>(collector.children.length);
```
Comment (iverase) [inline server/src/main/java/org/elasticsearch/search/profile/query/InternalProfileCollector.java:118]: I wonder if you need to check for null here now?

**Review by iverase — APPROVED** [review]
LGTM. Just have a small doubt on null checks but it is a nice change.

File: server/src/main/java/org/elasticsearch/search/profile/query/InternalProfileCollector.java (hunk 1 above)
Comment (iverase) [reply server/src/main/java/org/elasticsearch/search/profile/query/InternalProfileCollector.java:118]: forget this, just check and it get initialize, so this is fine.

Comment (javanna) [reply server/src/main/java/org/elasticsearch/search/profile/query/InternalProfileCollector.java:118]: yea, I think it's a good idea.

**Pull request merged**